src/libraries/System.Private.Xml/src/System/Xml/Schema/XsdBuilder.cs

```diff
@@ -793,7 +793,7 @@ internal override void StartChildren()
             {
                 if (_namespaces != null && _namespaces.Count > 0)
                 {
-                    _xso.Namespaces.Namespaces = _namespaces!;
+                    _xso.Namespaces = new XmlSerializerNamespaces(_namespaces.ToArray());
```

Comment:
Add an internal ctor that takes a **List\<XmlQualifiedName>** to avoid allocation an array?